motor/providers/terraform/provider.go

```diff
@@ -79,22 +85,36 @@ func New(tc *providers.Config) (*Provider, error) {
 		}
 
 		if stat.IsDir() {
-			fileList, err := os.ReadDir(path)
-			if err != nil {
-				return nil, err
-			}
-
-			err = loader.ParseHclDirectory(path, fileList)
-			if err != nil {
-				return nil, errors.Wrap(err, "could not parse hcl files")
-			}
-
-			err = ReadTfVarsFromDir(path, fileList, tfVars)
-			if err != nil {
-				return nil, errors.Wrap(err, "could not parse tfvars files")
-			}
-
-			modulesManifest, err = ParseTerraformModuleManifest(path)
+			filepath.WalkDir(path, func(path string, d fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+				foundExamples := MODULE_EXAMPLES.FindString(path)
+				if foundExamples != "" {
+					log.Debug().Str("path", path).Msg("ignoring terraform module example")
+					return nil
+				}
+				if !d.IsDir() {
+					log.Debug().Str("path", path).Msg("parsing hcl file")
+					err = loader.ParseHclFile(path)
+					if err != nil {
+						return errors.Wrap(err, "could not parse hcl file")
+					}
+
+					err = ReadTfVarsFromFile(path, tfVars)
+					if err != nil {
+						return errors.Wrap(err, "could not parse tfvars file")
+					}
+				} else {
+					modulesManifest, err = ParseTerraformModuleManifest(path)
```

Comment:
As I don't know which directory holds the `.terraform` subdirectory, I have to scan each directory for it. But it is no error, when it is not found.